Description: Fix build issues on Alpha
 This patch fixes build issues on Alpha by adding the necessary
 assembly routines for VT.
 .
 It can be removed when 1.3.4 is released.
Forwarded: https://svn.open-mpi.org/trac/ompi/ticket/1973
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=510845
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=517543
Author: Manuel Prinz <manuel@debian.org>
Last-Update: 2009-10-13

--- a/ompi/contrib/vt/vt/vtlib/vt_pform_linux.c
+++ b/ompi/contrib/vt/vt/vtlib/vt_pform_linux.c
@@ -170,6 +170,8 @@
 # elif defined(__ia64__)
     /* ... ITC */
     clock_value = __getReg(_IA64_REG_AR_ITC);
+# elif defined(__alpha__)
+    asm volatile ("rpcc %0" : "=r" (clock_value));
 # else
     /* ... TSC */
     {
--- a/opal/asm/base/ALPHA.asm
+++ b/opal/asm/base/ALPHA.asm
@@ -197,3 +197,28 @@
 	addl $31,$0,$0
 	ret $31,($26),1
 	.end opal_atomic_cmpset_rel_64
+	.align 2
+	.align 4
+	.globl opal_sys_timer_get_cycles
+	.ent opal_sys_timer_get_cycles
+$opal_sys_timer_get_cycles..ng:
+opal_sys_timer_get_cycles:
+	.eflag 48
+	.frame $30,0,$26,0
+	.prologue 0
+	.set	macro
+	wmb
+	1:  ldq_l $0, 0($16)     
+	cmpeq $0, $17, $0     
+	beq $0, 2f           
+	mov $18, $0           
+	stq_c $0, 0($16)         
+	beq $0, 1b           
+	jmp 3f               
+2:  mov $31, $0      
+3:                   
+
+	.set	nomacro
+	rpcc $0
+	ret
+	.end opal_sys_timer_get_cycles
--- a/opal/asm/generated/atomic-alpha-linux.s
+++ b/opal/asm/generated/atomic-alpha-linux.s
@@ -197,5 +197,30 @@
 	addl $31,$0,$0
 	ret $31,($26),1
 	.end opal_atomic_cmpset_rel_64
+	.align 2
+	.align 4
+	.globl opal_sys_timer_get_cycles
+	.ent opal_sys_timer_get_cycles
+$opal_sys_timer_get_cycles..ng:
+opal_sys_timer_get_cycles:
+	.eflag 48
+	.frame $30,0,$26,0
+	.prologue 0
+	.set	macro
+	wmb
+	1:  ldq_l $0, 0($16)     
+	cmpeq $0, $17, $0     
+	beq $0, 2f           
+	mov $18, $0           
+	stq_c $0, 0($16)         
+	beq $0, 1b           
+	jmp 3f               
+2:  mov $31, $0      
+3:                   
+
+	.set	nomacro
+	rpcc $0
+	ret
+	.end opal_sys_timer_get_cycles
 
 	.section	.note.GNU-stack,"",@progbits
